protocols/client/client.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// gRPC client wrapper
+
+package client
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/mdlayher/vsock"
+	"google.golang.org/grpc"
+
+	agentgrpc "github.com/kata-project/agent/protocols/grpc"
+)
+
+const (
+	unixSocketScheme  = "unix"
+	vsockSocketScheme = "vsock"
+)
+
+type AgentClient struct {
+	agentgrpc.HyperstartServiceClient
+	conn *grpc.ClientConn
+}
+
+type dialer func(string, time.Duration) (net.Conn, error)
+
+// Supported sock address formats are:
+//   - unix://<unix socket path>
+//   - vsock://<cid>:<port>
+//   - <unix socket path>
+func NewAgentClient(sock string) (*AgentClient, error) {
+	addr, err := parse(sock)
+	if err != nil {
+		return nil, err
+	}
+	dialOpts := []grpc.DialOption{grpc.WithInsecure(), grpc.WithTimeout(5 * time.Second)}
```

Comment:
Might be a good idea to "surface" this timeout by creating a global (`var`) for it. That would make it clearer and also allow the tests to modify the value to force failure scenarios.